client_builder.go

```diff
@@ -103,7 +105,35 @@ func clientInvoke(ctx context.Context, innerClient internal.InnerClient, invocat
 		},
 	})
 	if err != nil {
-		return nil, unmarshalError(err.Error())
+		err = unmarshalError(err.Error())
+		var e *DesktopSessionExpired
+		if errors.As(err, &e) {
+			oldInnerClient := *innerClient
+			var clientID *uint64
+			clientID, err = innerClient.Core.InitClient(ctx, innerClient.Config)
+			if err != nil {
+				return nil, err
+			}
+			innerClient = &internal.InnerClient{
+				ID:     *clientID,
+				Core:   oldInnerClient.Core,
+				Config: oldInnerClient.Config,
+			}
```

Comment:
Is there a reason we can't just mutate the ID field:

```
*innerClient.ID = *clientID
```

instead of using the `oldInnerClient` for re-assigning the entire inner client?

Reply:
Nice catch. This can indeed be simplified.

Reply:
Addressed with [47b6439](https://github.com/1Password/onepassword-sdk-go/pull/224/commits/47b64399899669f9eec0626adaa13e27acfe720d)